R/design_helper_functions.R

```diff
@@ -127,6 +127,14 @@ run_design_internal.default <- function(design, current_df=NULL, results=NULL, s
   
 }
 
+# for when the user sends a function that runs a design itself 
+#   to run_design (or simulate_design / diagnose_design above it)
+run_design_internal.function <- function(design) {
```

Comment:
should also update the generic on run_design_internal.default to be design, now that accept multiple types, and have default thow an error.